### ISSUE TITLE
"Dropping expired message" caused by activation diagnostics system

### ISSUE THREAD

**Issue by EdeMeijer** [dotnet/orleans#7046]
Hello,

I was having some issues with long running / blocking grain calls, and was seeing warnings about dropped messages every 5 seconds. After enabling debug logs for Orleans and digging in the source code, I found the following problem:

In `ActivationData.AnalyzeWorkLoad()`, diagnostics responses for pending requests are created with `messageFactory.CreateDiagnosticResponseMessage(message, ...)`. That function copies the original request's remaining TTL to the response's TTL header. When the grain call is running for more than 30 seconds, the diagnostic responses are dead on arrival, and immediately discarded by the `OutboundMessageQueue`.

Effective logs look like this (redacted a bit):

```
DBUG | Creating Request S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Sip.Messages.InboundSipMessageGrain/d6938eeb-6b44-41fd-813a-ae66fc6931f8@0e930d8d->S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+[- snip -]@26d7979a InvokeMethodRequest Netwerkplek.Cloud.Orleans.Telephony.Utils.ILock:ReleaseAsync #3245 status update with diagnostics ["[Activation: S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+[- snip -]-0x327D3413@26d7979a #GrainType=Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain Placement=PreferLocalPlacement State=Valid NonReentrancyQueueSize=0 EnqueuedOnDispatcher=0 InFlightCount=2 NumRunning=2 IdlenessTimeSpan=00:02:36.6259210 CollectionAgeLimit=00:01:01]", "TaskScheduler status: WorkItemGroup:Name=[Activation: S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+[- snip -]@26d7979a #GrainType=Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain Placement=PreferLocalPlacement State=Valid],WorkGroupStatus=Running. Currently QueuedWorkItems=0; Total Enqueued=6; Total processed=5; Quantum expirations=0;  Executing Task Id=4 Status=Running for 00:02:35.2786557.TaskRunner=ActivationTaskScheduler-593:Queued=0; Detailed context=<[Activation: S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+[- snip -]-0x327D3413@26d7979a #GrainType=Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain Placement=PreferLocalPlacement State=Valid NonReentrancyQueueSize=0 EnqueuedOnDispatcher=0 InFlightCount=2 NumRunning=2 IdlenessTimeSpan=00:02:36.6259382 CollectionAgeLimit=00:01:01 CurrentlyExecuting=Request S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Sip.Messages.InboundSipMessageGrain/c5c52cd8-227a-43b5-94f4-106792b36bd6@4df33c98->S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+[- snip -]@26d7979a InvokeMethodRequest Netwerkplek.Cloud.Orleans.Telephony.Utils.ILock:AcquireAsync #3116]>", "Message Request S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Sip.Messages.InboundSipMessageGrain/c5c52cd8-227a-43b5-94f4-106792b36bd6@4df33c98->S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+Transport/OnRequest/[ - snip -]@26d7979a InvokeMethodRequest Netwerkplek.Cloud.Orleans.Telephony.Utils.ILock:AcquireAsync #3116 was has been executing for 00:02:36.1003343.", "Interleaving message Request S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Sip.Messages.InboundSipMessageGrain/d6938eeb-6b44-41fd-813a-ae66fc6931f8@0e930d8d->S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+[- snip -]@26d7979a InvokeMethodRequest Netwerkplek.Cloud.Orleans.Telephony.Utils.ILock:ReleaseAsync #3245 has been executing for 00:02:35.2779246."] 


WARN | Dropping expired message Status Response S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Utils.LockGrain/0+[- snip -]@26d7979a->S192.168.16.6:11111:355051325*grn/Netwerkplek.Cloud.Orleans.Telephony.Sip.Messages.InboundSipMessageGrain/d6938eeb-6b44-41fd-813a-ae66fc6931f8@0e930d8d #3245 at phase Send 
```

So, effectively, the diagnostics reporting breaks as soon as the original request that is being reported about has timed out.

Orleans version = 3.4.1

**Comment by ReubenBond** [comment]
Good find, @EdeMeijer, are you willing to create a PR to fix it?

**Comment by EdeMeijer** [comment]
Sure, I can take a shot, but then I'd have to know how it was intended to work in the first place. My guess is that diagnostic responses don't really need any kind of TTL since they are a kind of fire-and-forget message that does not get another reply back. I don't know if that's possible, otherwise I could just use some arbitrary high value. What do you think?

**Comment by EdeMeijer** [comment]
Ok that wasn't so hard, since TTLs are optional I just cleared them for diagnostic updates.

**Comment by ReubenBond** [comment]
Fixed in #7049
